app/lib/server/methods/saveSettings.js

```diff
@@ -1,27 +1,26 @@
 import { Meteor } from 'meteor/meteor';
 import { Match, check } from 'meteor/check';
 
-import { hasPermission } from '../../../authorization';
+import { hasPermission, hasAllPermission } from '../../../authorization';
 import { settings } from '../../../settings';
 import { Settings } from '../../../models';
 
 Meteor.methods({
 	saveSettings(params = []) {
+		const settingsNotAllowed = [];
 		if (Meteor.userId() === null) {
 			throw new Meteor.Error('error-action-not-allowed', 'Editing settings is not allowed', {
 				method: 'saveSetting',
 			});
 		}
 
-		if (!hasPermission(Meteor.userId(), 'edit-privileged-setting')) {
-			throw new Meteor.Error('error-action-not-allowed', 'Editing settings is not allowed', {
```

Comment:
This actually failed the saving of multiple settings on the first permission violation. Since now multiple settings can be saved at once and some may fail while others wont, I moved this error handling into a loop down below.